Mathlib/Data/Int/Interval.lean

```diff
@@ -129,6 +129,19 @@ theorem card_Ioc_of_le (h : a ≤ b) : (#(Ioc a b) : ℤ) = b - a := by
 theorem card_Ioo_of_lt (h : a < b) : (#(Ioo a b) : ℤ) = b - a - 1 := by
   rw [card_Ioo, sub_sub, toNat_sub_of_le h]
 
+theorem Icc_eq_pair (h : a + 1 = b) : Icc a b = {a, b} := by
```

Comment:
It's pretty uncommon to have a hypothesis like this. I would first add the version with `b` substituted, and if you still need this formulation, prove it as `subst h; exact other_lemma ..`.

Also golf:
```
example (a : Int) : Finset.Icc a (a + 1) = {a, a + 1} := by
  ext
  simp
  omega
```

Reply:
Done, thanks.